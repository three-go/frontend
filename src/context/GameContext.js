import React, { createContext, useState } from "react";

const GameContext = createContext();

export const GameContextProvider = ({ children }) => {
  const [currentGameKey, setCurrentGameKey] = useState("");

  return (
    <GameContext.Provider value={{ currentGameKey, setCurrentGameKey }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContext;
