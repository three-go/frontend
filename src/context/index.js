import GameContext, { GameContextProvider } from "./GameContext";

export { GameContext, GameContextProvider };
