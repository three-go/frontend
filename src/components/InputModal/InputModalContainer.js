import React, { useState } from "react";

import { Keyboard } from "react-native";

import { SmallButton, InputModal } from "../../components";

const InputModalContainer = () => {
  const [modalVisible, setModalVisible] = useState(true);
  const [name, setName] = useState("");

  // 유저 스코어 로직 가져오는 로직 필요

  const onRegister = (name) => {
    // 유저 스코어 등록 로직 필요
    setModalVisible(false);
    setName("");
    Keyboard.dismiss();
  };

  return (
    <InputModal
      score="200"
      modalVisible={modalVisible}
      name={name}
      setName={setName}
      onRegister={onRegister}
    >
      <SmallButton content="등록" color="#2196F3" onPress={onRegister} />
    </InputModal>
  );
};

export default InputModalContainer;
