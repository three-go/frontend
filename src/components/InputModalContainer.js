import React, { useState } from "react";

import { Keyboard } from "react-native";

import InputModal from "./InputModal";

const InputModalContainer = () => {
  const [modalVisible, setModalVisible] = useState(true);
  const [name, setName] = useState("");

  // 유저 스코어 로직 가져오는 로직 필요

  const onRegister = (name) => {
    // 유저 스코어 등록 로직 필요
    setModalVisible(false);
    setName("");
    Keyboard.dismiss();
  };

  return (
    <InputModal
      score="200"
      modalVisible={modalVisible}
      setModalVisible={setModalVisible}
      name={name}
      onRegister={onRegister}
    />
  );
};

export default InputModalContainer;
