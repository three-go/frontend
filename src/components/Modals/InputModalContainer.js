import React, { useState, useContext } from "react";

import { useNavigation } from "@react-navigation/native";
import PropTypes from "prop-types";
import { Keyboard } from "react-native";
import uuid from "react-native-uuid";

import { navigations, colors } from "../../common/constants";
import GameContext from "../../context/GameContext";
import {
  setItemToAsync,
  getItemFromAsync,
} from "../../utils/asyncStorageHelper";
import SmallButton from "../Buttons/SmallButton";
import InputModal from "./InputModal";

const InputModalContainer = ({ score }) => {
  const navigation = useNavigation();
  const { currentGameKey } = useContext(GameContext);
  const [modalVisible, setModalVisible] = useState(true);
  const [userName, setUserName] = useState("");

  const onRegister = async () => {
    try {
      const data = await getItemFromAsync(currentGameKey);

      if (!data) {
        await setItemToAsync(currentGameKey, [
          {
            id: uuid.v4(),
            name: userName.trim() === "" ? "이름없음" : userName,
            score,
          },
        ]);
      } else {
        data.push({
          id: uuid.v4(),
          name: userName.trim() === "" ? "이름없음" : userName,
          score,
        });
        await setItemToAsync(currentGameKey, data);
      }

      setModalVisible(false);
      Keyboard.dismiss();
      navigation.navigate(navigations.main, { visible: true });
    } catch (error) {
      setModalVisible(false);
      Keyboard.dismiss();
      navigation.navigate(navigations.main);
    }
  };

  return (
    <InputModal
      score={score}
      modalVisible={modalVisible}
      name={userName}
      setName={setUserName}
      onRegister={onRegister}
    >
      <SmallButton
        content="등록"
        color={colors.tealGreen}
        onPress={onRegister}
      />
    </InputModal>
  );
};

export default InputModalContainer;

InputModalContainer.propTypes = {
  score: PropTypes.number.isRequired,
};
