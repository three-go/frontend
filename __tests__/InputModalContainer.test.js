import React from "react";

import { render } from "@testing-library/react-native";

import InputModalContainer from "../src/components/Modals/InputModalContainer";
import { GameContextProvider } from "../src/context/GameContext";

const mockNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

jest.mock("@react-native-community/async-storage", (key, value) => {
  const storage = {
    setItemToAsync: () => {},
    getItemFromAsync: () => {},
  };
  storage[key] = value;
  return storage;
});

describe("InputModalContainer", () => {
  it("InputModalContainer render check", () => {
    const score = 100;
    const component = render(
      <GameContextProvider>
        <InputModalContainer score={score} />
      </GameContextProvider>
    );

    expect(component.getByText(new RegExp(score, "i"))).toBeTruthy();
    expect(component.toJSON).toMatchSnapshot();
    expect(component.toJSON).toBeTruthy();
  });
});
