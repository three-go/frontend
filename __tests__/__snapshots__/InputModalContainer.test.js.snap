// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputModalContainer InputModalContainer render check 1`] = `[Function]`;
